models/darknet/darknet.hpp

```diff
@@ -0,0 +1,311 @@
+   *
+   * @param inputShape A three-valued tuple indicating input shape.
+   *                   First value is number of Channels (Channels-First).
+   *                   Second value is input height.
+   *                   Third value is input width..
+   * @param numClasses Optional number of classes to classify images into,
+   *                   only to be specified if includeTop is  true.
+   * @param weights One of 'none', 'cifar10'(pre-training on CIFAR10) or path to weights.
+   */
+  DarkNet(const std::tuple<size_t, size_t, size_t> inputShape,
+          const size_t numClasses = 1000,
+          const std::string& weights = "none",
+          const bool includeTop = true);
+
+  //! Get Layers of the model.
+  FFN<OutputLayerType, InitializationRuleType>& GetModel() { return darkNet; }
+
+  //! Load weights into the model.
+  void LoadModel(const std::string& filePath);
+
+  //! Save weights for the model.
+  void SaveModel(const std::string& filePath);
+
+ private:
+  /**
+   * Adds Convolution Block.
+   *
+   * @tparam SequentialType Layer type in which convolution block will
+   *                        be added.
+   *
+   * @param inSize Number of input maps.
+   * @param outSize Number of output maps.
+   * @param kernelWidth Width of the filter/kernel.
+   * @param kernelHeight Height of the filter/kernel.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   * @param batchNorm Boolean to determine whether a batch normalization
+   *                  layer is added.
+   * @param baseLayer Layer in which Convolution block will be added, if
+   *                  NULL added to darkNet FFN.
+   */
+  template<typename SequentialType = Sequential<>>
+  void ConvolutionBlock(const size_t inSize,
+                        const size_t outSize,
+                        const size_t kernelWidth,
+                        const size_t kernelHeight,
+                        const size_t strideWidth = 1,
+                        const size_t strideHeight = 1,
+                        const size_t padW = 0,
+                        const size_t padH = 0,
+                        const bool batchNorm = false,
+                        SequentialType* baseLayer = NULL)
+  {
+    Sequential<>* bottleNeck = new Sequential<>();
+    bottleNeck->Add(new Convolution<>(inSize, outSize, kernelWidth,
+        kernelHeight, strideWidth, strideHeight, padW, padH, inputWidth,
+        inputHeight));
+    if (batchNorm)
+    {
+      bottleNeck->Add(new BatchNorm<>(outSize));
+    }
+
+    bottleNeck->Add(new LeakyReLU<>());
+
+    // Update inputWidth and input Height.
+    std::cout << "Conv Layer.  ";
+    std::cout << "(" << inputWidth << ", " << inputHeight <<
+        ", " << inSize << ") ----> ";
```

Comment:
Do you think we should use `mlpack::Log` here instead?

Reply:
Sure, I'll make the changes.